app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,46 @@
+# Comments model belongs to book and to user.
+class CommentsController < ApplicationController
+  def show
+    comment = Comment.find(params[:id])
+    render json: comment, status: :ok
+  end
+
+  def index
+    comments = Comment.where(book_id: params[:book_id])
+    render json: comments, status: :ok
+  end
+
+  def create
+    authorize :comment
+    book = Book.find(params[:book_id])
+    comment = book.comments.build(comments_params)
+    comment.user_id = current_user.id
+    if comment.save
+      render json: comment, status: :ok
+    else
+      render json: { errors: comment.errors }, status: :unprocessable_entity
+    end
+  end
+
+  def update
+    authorize :comment
```

Comment:
you should pass the comment in this case because only the author or an admin should be able to edit a comment.
Create a private method like this
`def comment`
`    @comment ||= Comment.where(book_id: params[:book_id])`
`end`

to access the comment